safety/scan/main.py

```diff
@@ -196,8 +203,43 @@ def save_report_as(scan_type: ScanType, export_type: ScanExport, at: Path, repor
     with open(at, 'w+') as report_file:
         report_file.write(report)
 
+def build_meta(target: Path) -> Dict[str, Any]:
+    """
+    Build the meta JSON object for a file.
+
+    Args:
+        target (Path): The path of the repository.
 
-def process_files(paths: Dict[str, Set[Path]], config: Optional[ConfigModel] = None) -> Generator[Tuple[Path, InspectableFile], None, None]:
+    Returns:
+        Dict[str, Any]: The metadata dictionary.
+    """
+    target_obj = target.resolve()
+    git_utils = GIT(target_obj)
+
+    git_data = git_utils.build_git_data()
+    git_metadata = {
+        "git_branch": git_data.branch if git_data else None,
+        "git_commit": git_data.commit if git_data else None,
+        "dirty": git_data.dirty if git_data else None,
+        "tag": git_data.tag if git_data else None,
+        "git_origin": git_data.origin if git_data else None,
+    }
+
+    os_metadata = {
+        "os_type": os.environ.get("SAFETY_OS_TYPE", None) or platform.system(),
```

Comment:
Similar to above the fields under `os` section does not need to have `os_` prefix.